packages/twenty-server/src/engine/api/rest/core/rest-api-core-v2.service.ts

```diff
@@ -0,0 +1,63 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+
+import { Request } from 'express';
+import { capitalize } from 'twenty-shared';
+
+import { CoreQueryBuilderFactory } from 'src/engine/api/rest/core/query-builder/core-query-builder.factory';
+import { parseCorePath } from 'src/engine/api/rest/core/query-builder/utils/path-parsers/parse-core-path.utils';
+import { extractObjectIdFromPath } from 'src/engine/api/rest/core/utils/extract-id-from-path.utils';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+
+@Injectable()
+export class RestApiCoreServiceV2 {
+  constructor(
+    private readonly coreQueryBuilderFactory: CoreQueryBuilderFactory,
+    private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+  ) {}
+
+  async delete(request: Request) {
+    const { workspace } = request;
+    const { object: parsedObject } = parseCorePath(request);
+    const objectId = extractObjectIdFromPath(request.path);
+
+    const objectMetadata = await this.coreQueryBuilderFactory.getObjectMetadata(
+      request,
+      parsedObject,
+    );
+
+    if (!objectMetadata) {
+      throw new BadRequestException('Object not found');
+    }
+
+    const objectMetadataNameSingular =
+      objectMetadata.objectMetadataItem.nameSingular;
+
+    if (!workspace?.id) {
+      throw new BadRequestException('Workspace not found');
+    }
+
+    const repository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace(
+        workspace.id,
+        objectMetadataNameSingular,
+      );
+
+    const itemToDelete = await repository.findOne({
+      where: {
+        id: objectId,
```

Comment:
Globally on the repository, `object` refers to an entity like Company or Person. `record` refers to an instance of that entity, like Rbnb or Tim Cook.

Delete is about deleting a record, not an object. So I would rename like:
```suggestion
        id: recordId,
```